### DISCUSSION TITLE
Using Colormaps in my ImGui application

### DISCUSSION THREAD

**Discussion by JamAndCheese** [epezent/implot#209] [Q&A] [answered]
Hello, (I hope this is the right place to ask this. If not, please direct me to the right place or feel free to move it.)

I have an ImGui application that I wrote to display data bitmaps, single-channel normalized data (0.0 to 1.0), form various sources. I implemented my own colormaps (which included a Jet-like palette). However, seeing your amazing work on Colormaps, I would like to include ImPlot just to leverage this. Is this possible?  

What I would need is a transformation of my 0.0 - 1.0 values into an RGB8 triplet or an equivalent ImU32, in which I will ignore the Alpha components. I assume that one of your available colormaps would have been selected previously. (I will probably make them all available for user selection)

Looking forward to any ideas or comments!

**Comment by epezent** [comment, marked as answer]
We have `SampleColormap` which returns an ImVec4 given t [0 1]. We could expose an ImU32 version if that's what you are asking for.

**Reply by JamAndCheese** [reply]
That would be awesome if you do. However I will start playing with SampleColormap and convert the floats for now. Thank you for taking the time to respond!

**Reply by JamAndCheese** [reply]
@epezent I found a bug in your **SampleColormapU32** function. See the original **return** commented out. It was not using the argument cmap, therefore it was always using the current style one.

    ImU32  SampleColormapU32(float t, ImPlotColormap cmap) {
        ImPlotContext& gp = *GImPlot;
        cmap = cmap == IMPLOT_AUTO ? gp.Style.Colormap : cmap;
        IM_ASSERT_USER_ERROR(cmap >= 0 && cmap < gp.ColormapData.Count, "Invalid colormap index!");
        //    return gp.ColormapData.LerpTable(gp.Style.Colormap,t);
        return gp.ColormapData.LerpTable( cmap, t );
    }

**Reply by epezent** [reply]
Fixed. I suppose you discovered you can include `implot_internal.h` and use the ImU32 version now. My point above was in regards to exposing this in the "official" API (i.e. `implot.h`).

**Reply by JamAndCheese** [reply]
Thank you!
